pkg/component/kubestatemetrics/resources.go

```diff
@@ -387,6 +391,44 @@ func (k *kubeStateMetrics) reconcileServiceMonitor(serviceMonitor *monitoringv1.
 	}
 }
 
+func (k *kubeStateMetrics) emptySeedScrapeConfig() *monitoringv1alpha1.ScrapeConfig {
+	return &monitoringv1alpha1.ScrapeConfig{ObjectMeta: monitoringutils.ConfigObjectMeta("kube-state-metrics", k.namespace, seed.Label)}
+}
+
+func (k *kubeStateMetrics) reconcileSeedScrapeConfig(scrapeConfig *monitoringv1alpha1.ScrapeConfig) {
+	scrapeConfig.Labels = monitoringutils.Labels(seed.Label)
+	scrapeConfig.Spec = monitoringv1alpha1.ScrapeConfigSpec{
+		KubernetesSDConfigs: []monitoringv1alpha1.KubernetesSDConfig{{
+			Role:       "service",
+			Namespaces: &monitoringv1alpha1.NamespaceDiscovery{Names: []string{k.namespace}},
+		}},
+		RelabelConfigs: []*monitoringv1.RelabelConfig{
+			{
+				SourceLabels: []monitoringv1.LabelName{
+					"__meta_kubernetes_service_label_component",
+					"__meta_kubernetes_service_port_name",
+				},
+				Regex:  "kube-state-metrics;" + portNameMetrics,
+				Action: "keep",
+			},
+			{
+				Action:      "replace",
+				Replacement: "kube-state-metrics",
+				TargetLabel: "job",
+			},
```

Comment:
It is needed when using `ScrapeConfig`s. When using `ServiceMonitor`s, `prometheus-operator` uses the service name as job name, however, since there is no service definition, it cannot do so with `ScrapeConfig`s. Hence, we have to set the `job` label manually.